packages/shared/src/utils/getAssetSelector.ts

```diff
@@ -0,0 +1,29 @@
+import { keccak256 } from '@ethersproject/solidity'
+
+import { AssetType } from '../AssetDetails'
+import { assertUnreachable } from './assert'
+
+export function getAssetSelector(
+  assetType:
+    | Extract<AssetType, 'ETH' | 'ERC20'>
+    | Extract<AssetType, 'ERC721' | 'ERC1155'>
```

Comment:
Couldn't we define it like this `Extract<AssetType, 'ERC721' | 'ERC1155' | 'ETH' | 'ERC20'>`

Reply:
yup, we should